pkg/app/pipedv1/plugin/platform/kubernetes/planner/server.go

```diff
@@ -63,7 +77,39 @@ func (ps *PlannerService) DetermineStrategy(ctx context.Context, in *platform.De
 }
 
 func (ps *PlannerService) QuickSyncPlan(ctx context.Context, in *platform.QuickSyncPlanRequest) (*platform.QuickSyncPlanResponse, error) {
-	return nil, fmt.Errorf("not implemented yet")
+	now := time.Now()
+
+	cloner, err := plugin.GetPlanSourceCloner(in.GetInput())
+	if err != nil {
+		return nil, err
+	}
+
+	d, err := os.MkdirTemp("", "") // TODO
+	if err != nil {
+		return nil, fmt.Errorf("failed to prepare temporary directory (%w)", err)
+	}
+	defer os.RemoveAll(d)
+
+	p := deploysource.NewProvider(
+		d,
+		cloner,
+		*in.GetInput().GetDeployment().GetGitPath(),
+		ps.Decrypter,
+	)
+
+	ds, err := p.GetReadOnly(ctx, io.Discard /* TODO */)
+	if err != nil {
+		return nil, err
+	}
+
+	autoRollback := false
+	if a := ds.ApplicationConfig.KubernetesApplicationSpec.Input.AutoRollback; a != nil {
+		autoRollback = *a
+	}
```

Comment:
nits, we have a default value set as true, so no need to initialize autoRollback var as false here, since the `ds.ApplicationConfig.KubernetesApplicationSpec.Input.AutoRollback` is true as init
ref: ds.ApplicationConfig.KubernetesApplicationSpec.Input.AutoRollback

Reply:
Thank you.
I noticed that I can view the document here.
https://pkg.go.dev/github.com/pipe-cd/pipecd/pkg/config#KubernetesDeploymentInput

Reply:
I fixed it on this commit.
https://github.com/pipe-cd/pipecd/pull/5020/commits/d845f7a1e293dea974f1544db818dff83fbe5531